CONTRIBUTING.md

```diff
@@ -149,3 +140,56 @@ This repository also includes [stylecop ruleset
 files](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/master/build).
 These files are used to configure the _StyleCop.Analyzers_ which runs during
 build. Breaking the rules will result in a build failure.
+
+## Contributing a new project
+
+This repo is a great place to contribute a new instrumentation, exporter or
+any kind of extension. Please refer to
+[this page](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/extending-the-sdk/README.md#extending-the-opentelemetry-net-sdk)
+for help writing your component.
+Although the projects within this repo share some
+properties and configurations, they are built and released independently.
+So if you are creating a new project within `/src` and corresponding test
+project within `/test`, here are a few things you should do to ensure that
+your project is automatically built and shipped through CI.
+
+* Based on what your project is, you may need to depend on the
+[OpenTelemetry SDK](https://www.nuget.org/packages/OpenTelemetry)
+or the
+[OpenTelemetry API](https://www.nuget.org/packages/OpenTelemetry.Api)
+Include the necessary package in your project.
+You can choose the version that you want to depend on.
+Usually it is a good idea to use the latest version. Example:
+
+  ```xml
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry" Version="1.0.0-rc4" />
+  </ItemGroup>
+  ```
+
+* The assembly and nuget versioning is managed through
```

Comment:
probably too much info here, as the actual version is based on git tag, which is good candidate to be put in RELEASING.md file (once we have one).
For contributing.md, we can simply state "your project has minvertagprefix which matches project name", and an example.

Reply:
+1 for having a RELEASING.md file and having most of the info about versioning there.